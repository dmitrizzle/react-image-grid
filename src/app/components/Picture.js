import React from "react"
import styled from "styled-components"

export default styled.div`
  width: 225px;
  height: 225px;
  box-shadow: 0 0 1px #999 inset;
`
