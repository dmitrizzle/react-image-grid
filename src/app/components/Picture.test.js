import "jest-styled-components";

import { Sugar } from "@roast-cms/react-sugar-styled";
import React from "react";

import { shallow } from "enzyme";

import Picture, { PictureWrap, Image, PictureOverlay } from "./Picture";

test("Render PictureWrap without crashing, match snapshot", () => {
  const element = shallow(<PictureWrap theme={Sugar({})} />);
  expect(element).toMatchSnapshot();
});
test("Render Picture without crashing, match snapshot", () => {
  const element = shallow(<Picture />);
  expect(element).toMatchSnapshot();
});
test("Render Image without crashing, match snapshot", () => {
  const element = shallow(<Image />);
  expect(element).toMatchSnapshot();
});
test("Render PictureOverlay without crashing, match snapshot", () => {
  const element = shallow(<PictureOverlay theme={Sugar({})} />);
  expect(element).toMatchSnapshot();
});
