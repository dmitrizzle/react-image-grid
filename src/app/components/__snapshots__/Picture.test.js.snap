// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Image without crashing, match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-bwzfXH eHOWqo",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-bwzfXH eHOWqo",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render Picture without crashing, match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styled.div />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styled.div />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Render PictureWrap without crashing, match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div
    theme={
      Object {
        "color": Object {
          "background": [Function],
          "brand": [Function],
          "foreground": [Function],
          "highlight": [Function],
        },
        "effects": Object {
          "borderRadius": Object {
            "med": 0.45,
            "small": [Function],
          },
        },
        "layer": Object {
          "card": 30,
          "nav": 20,
          "overlay": 40,
          "tuck": -1,
          "up": 10,
        },
        "opacity": Object {
          "half": 0.5,
          "least": 0.125,
          "most": 0.85,
        },
        "size": Object {
          "block": Object {
            "border": 4,
            "column": Object {
              "l": 900,
              "m": 700,
            },
            "padding": 1.5,
            "spacing": 0.75,
          },
          "breakpoint": Object {
            "exact": Object {
              "l": [Function],
              "m": [Function],
              "s": [Function],
              "xl": [Function],
              "xs": [Function],
              "xxl": [Function],
            },
            "max": Object {
              "l": [Function],
              "m": [Function],
              "s": [Function],
              "xl": [Function],
              "xs": [Function],
              "xxl": [Function],
            },
            "min": Object {
              "l": [Function],
              "m": [Function],
              "s": [Function],
              "xl": [Function],
              "xs": [Function],
              "xxl": [Function],
            },
            "stops": Object {
              "max": 1600,
              "min": 320,
            },
          },
          "font": Object {
            "auto": Array [
              "
        ",
              "
		@media (min-width: ",
              "32.5625",
              "em) {
			",
              "font-size: 	",
              [Function],
              "px;",
              "
		}
	",
              " ",
              "
		@media (max-width: ",
              "32.5",
              "em) {
			",
              "font-size: 	",
              [Function],
              "px;",
              "
		}
	",
              " ",
              "
		@media (max-width: ",
              "20",
              "em) {
			",
              "font-size: 	",
              [Function],
              "px;",
              "
		}
	",
              " ",
              "
		@media (min-width: ",
              "100.0625",
              "em) {
			",
              "font-size:	",
              [Function],
              "px;",
              "
		}
	",
              ";
      ",
            ],
            "l": 20.7,
            "m": 18,
            "make": Object {
              "larger": 2,
              "normal": 1,
              "smaller": 0.85,
              "tiny": 0.5,
            },
            "s": 16.2,
            "xs": 15.299999999999999,
          },
        },
        "typography": Object {
          "text": Object {
            "auto": Array [
              "
          color: ",
              [Function],
              ";
          font-family: ",
              "Arial, sans-serif",
              ";
          letter-spacing: 0.025em;
          line-height: ",
              [Function],
              "em;
        ",
            ],
            "font": "Arial, sans-serif",
            "lineHeight": 1.75,
          },
          "title": Object {
            "auto": Array [
              "
          color: ",
              [Function],
              ";
          font-family: ",
              "Arial, sans-serif",
              ";
          letter-spacing: 0.025em;
          line-height: ",
              [Function],
              "em;
          font-weight: ",
              "700",
              ";
          margin: 0;
        ",
            ],
            "font": "Arial, sans-serif",
            "lineHeight": 1.15,
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-bdVaJa bjmXyT",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-bdVaJa bjmXyT",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
